libultraship/libultraship/SDLController.cpp

```diff
@@ -229,11 +233,12 @@ namespace Ship {
                 NegButton == BTN_VSTICKUP || NegButton == BTN_VSTICKDOWN)) {
 
                 // The axis is being treated as a "button"
-                if (AxisValue > AxisMinimumPress) {
+                auto normalizedAxisValue = NormaliseStickValue(AxisValue);
+                if (normalizedAxisValue > AxisDeadzone) {

```

Comment:
Why change this to the deadzone when the axis is being used as a button? Otherwise the call to normalize stick value makes sense to me. We should be checking deadzones and presses by the n64 range, not the physical (sdl) range.

Reply:
Not quite sure I understand the question here sorry! But I'll explain why I've done it this way anyway to hopefully answer your question 🤞 

A trace log that I added:
`AxisValue: 24030 || AxisMinimumPress: 7680 || AxisDeadzone: 57`

From what I could see based on logs, the `AxisValue` scales with the full range `32767` (what you are calling the n64 range I guess?) as does the `AxisMinimumPress`, however the `AxisMinimumPress` is not affected by the deadzone. So then it was a choice of either scale the AxisMinimumPress by the deadzone to give a minimumPress value which accommodates the deadzone, or to scale the `AxisValue` to the SDL range so that the deadzone is correctly applied.

Based on what was already being done in the `NormalizeStickAxis` code (which is called in the same codeblock but for non-button based axis) we already have a deadzone implementation that scales the AxisValue to the SDL range and applies accordingly.

See https://github.com/HarbourMasters/Shipwright/blob/develop/libultraship/libultraship/SDLController.cpp#L60 :
```cpp
    void SDLController::NormalizeStickAxis(SDL_GameControllerAxis axisX, SDL_GameControllerAxis axisY, int16_t axisThreshold, int32_t virtualSlot) {
        auto profile = getProfile(virtualSlot);

        const auto axisValueX = SDL_GameControllerGetAxis(Cont, axisX);
        const auto axisValueY = SDL_GameControllerGetAxis(Cont, axisY);

        //scale {-32768 ... +32767} to {-84 ... +84}
        auto ax = axisValueX * 85.0f / 32767.0f;
        auto ay = axisValueY * 85.0f / 32767.0f;

        //create scaled circular dead-zone in range {-15 ... +15}
        auto len = sqrt(ax * ax + ay * ay);
        if (len < axisThreshold) {
            len = 0.0f;
        }
        else if (len > 85.0) {
            len = 85.0f / len;
        }
        else {
            len = (len - axisThreshold) * 85.0f / (85.0f - axisThreshold) / len;
        }
        ax *= len;
        ay *= len;
```

Of course, when calculating for button presses we don't need to worry about creating a radial deadzone since the button mappings are 2-dimensional, so to avoid repeating the complex logic above we can just check if the stick value falls outside the deadzone on that axis.

Reply:
Well I like the idea of doing all calculations in the N64 range, which this PR does seem to change, but it removes the usage of "AxisMinimumPress" completely. If this is the route to take, then we want to remove it from configuration as well.

Reply:
I think what we actually want is to use the "AxisMinimumPress" still, but use the Normalized value like you do here.

Reply:
We can also do >= AxisDeadzone && >=AxisMinimumValue

and we want AxisMinimumValue >= -85 <= 85